### PR TITLE
Main

### PR DESCRIPTION
**Pull request by Mozzi22**

### CODE REVIEW

File: src/Store/modals/types.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export type TModalData = {
+    type: string;
+    data: {};
+    isOpen: boolean;
+}
+
+export type TInitialModalData = {
+    createGame: TModalData;
+}
```
Comment (MaxShvachko) [inline src/Store/modals/types.ts:9]: add empty space after this line

File: src/Store/modals/types.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export type TModalData = {
+    type: string;
```
Comment (MaxShvachko) [inline src/Store/modals/types.ts:2]: create type TModalType = "createGame";

and pass it into this field

File: src/components/Main/RoomList/ControlPanel/ControlPanel.tsx [hunk 3]
```diff
@@ -34,8 +34,10 @@ const ControlPanel: React.FC<IControlPanel> = ({ setValue, filterByRoomName, cha
                     name='create_room'
                     onClick={handleCreateRoom}
                     content={t('create_room')}
-                    width="50px"
-                    margin="15px 0"
+                    width="100px"
+                    height="60px"
+                    backgroundSize='0'
+
```
Comment (MaxShvachko) [inline src/components/Main/RoomList/ControlPanel/ControlPanel.tsx:40]: delete empty line

File: src/components/Main/RoomList/ControlPanel/ControlPanel.tsx [hunk 4]
```diff
@@ -34,8 +34,10 @@ const ControlPanel: React.FC<IControlPanel> = ({ setValue, filterByRoomName, cha
                     name='create_room'
                     onClick={handleCreateRoom}
                     content={t('create_room')}
-                    width="50px"
-                    margin="15px 0"
+                    width="100px"
+                    height="60px"
+                    backgroundSize='0'
```
Comment (MaxShvachko) [inline src/components/Main/RoomList/ControlPanel/ControlPanel.tsx:39]: by default props backgroundSize must be '0' 

create default props value in component Button and remove this line

File: src/components/Main/RoomList/ControlPanel/index.ts [hunk 5]
```diff
@@ -1,3 +1,11 @@
+import { connect } from 'react-redux';
 import ControlPanel from './ControlPanel';
+// import { rooms } from '/src/Store/chat/selectors';
```
Comment (MaxShvachko) [inline src/components/Main/RoomList/ControlPanel/index.ts:3]: ??? 
if you don't need this comment. Remove it

File: src/components/Main/RoomList/ControlPanel/index.ts [hunk 6]
```diff
@@ -1,3 +1,11 @@
...
+import { changeModalVisibility } from 'src/Store/modals/actions';
 
-export default ControlPanel;
+// const mapStateToProps = state => ({ rooms: rooms(state) });
```
Comment (MaxShvachko) [inline src/components/Main/RoomList/ControlPanel/index.ts:6]: the same

File: src/components/Main/RoomList/RoomList.tsx [hunk 7]
```diff
@@ -12,13 +12,13 @@ const RoomList: React.FC<TRoomListItems> = ({ rooms = [] }) => {
                 <ControlPanel />
             </StControlPanel>
             <StItems>
-                {rooms?.length > 0 ? rooms.map(room => (
+                {/* {rooms?.length > 0 ? rooms.map(room => ( */}
```
Comment (MaxShvachko) [inline src/components/Main/RoomList/RoomList.tsx:15]: Check if need this comments in your pull request

File: src/components/Main/RoomList/RoomListItems/RoomListItems.tsx [hunk 8]
```diff
@@ -2,49 +2,67 @@ import React, { useEffect, useState } from 'react';
...
+    const handleJoinToGameWithBot = () => {
+    };
+    const handleClick = () => {
+      setValue({ name: 'currentRoom', value: { room_id: id, room_name: login } });
+    };
+    useEffect(() => {
+      if (currentRoomName === login) {
+        return setState({ ...state, selected: true });
+      }
+      setState({ ...state, selected: false });
```
Comment (MaxShvachko) [inline src/components/Main/RoomList/RoomListItems/RoomListItems.tsx:29]: all React Hooks must be on top in your components and create setState before useEffect

File: src/components/Main/RoomList/RoomListItems/RoomListItems.tsx [hunk 9]
```diff
@@ -2,49 +2,67 @@ import React, { useEffect, useState } from 'react';
...
+                          id={el.id}
+                          key={el.id}
+                          name={el.id}
+                          content={t(el.id)}
+                          width="200px"
+                          height="50px"
+                          margin="15px 0"
+                          onClick={getFunctionForButtons(el)}
+                          backgroundSize='0'
+
```
Comment (MaxShvachko) [inline src/components/Main/RoomList/RoomListItems/RoomListItems.tsx:60]: empty line

File: src/components/UI/Modal/ModalComponent.tsx [hunk 10]
```diff
@@ -36,6 +36,8 @@ const ModalComponent = ({
                         id="closeModal"
                         onClick={handleCloseModal}
                         bgColor="transparent"
+                        backgroundSize='0'
+
```
Comment (MaxShvachko) [inline src/components/UI/Modal/ModalComponent.tsx:40]: line

File: src/components/UI/Modal/types.ts [hunk 11]
```diff
@@ -4,7 +4,7 @@ export interface IModalComponent {
     isOpen: boolean;
     changeModalVisibility: (e : React.MouseEvent<HTMLElement, MouseEvent>) => void;
     headerTextKey: string;
-    modalType: string;
+    modalType: any;
```
Comment (MaxShvachko) [inline src/components/UI/Modal/types.ts:7]: pass type TModalType into modalType

File: src/index.js [hunk 12]
```diff
@@ -9,14 +9,14 @@ import App from './components/App';
 import rootSaga from './Store/saga';
 import './i18n';
 
-const devTools = window?.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
+// const devTools = window?.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
```
Comment (MaxShvachko) [inline src/index.js:12]: please return devTools. We need it )

File: src/components/Main/RoomList/RoomList.tsx (hunk 7 above)
Comment (Mozzi22) [reply src/components/Main/RoomList/RoomList.tsx:15]: это проверка, если у нас нету игр отобразить текст по типу "нету доступных игр, создайте"

**Pull request merged**